server/src/main.py

```diff
@@ -96,6 +98,13 @@ def manage_durable_session():
         logging.error(e)
         logout_user()
 
+  @app.after_request
+  def apply_csp(response):
+    # this isn't necessary for Okta or OneLogin, but Azure is more pedantic
+    response.headers['Content-Security-Policy'] = "default-src 'self' data:; style-src 'self' fonts.googleapis.com 'unsafe-inline'; font-src fonts.gstatic.com"
```

Comment:
Im not sure about `'unsafe-inline'`:
Can we rerun pentest and see if it works?

The right solution here is generate hash for each inline style and put it inside CSP for particular user.